divi/src/script/standard.cpp

```diff
@@ -148,6 +160,38 @@ bool ExtractScriptPubKeyFormat(const CScript& scriptPubKey, txnouttype& typeRet,
                 else
                     break;
             }
+            else if (opcode2 == OP_ANYHASH)
+            {
+                bool valid;
+                switch (opcode1)
+                {
+                case OP_RIPEMD160:
+                case OP_SHA1:
+                case OP_SHA256:
+                case OP_HASH160:
+                case OP_HASH256:
+                    valid = true;
+                    vSolutionsRet.push_back(valtype(1, opcode1));
+                    break;
+
+                default:
+                    valid = false;
+                    break;
+                }
+                if (!valid)
+                    break;
+            }
+            else if (opcode2 == OP_ANY)
+            {
+                if (opcode1 == OP_0 ||
+                    (opcode1 >= OP_1 && opcode1 <= OP_16))
+                {
+                    char n = (char)CScript::DecodeOP_N(opcode1);
+                    vSolutionsRet.push_back(valtype(1, n));
+                }
+                else
+                    vSolutionsRet.push_back(vch1);
+            }
```

Comment:
I actually had this with just SHA256 initially.  But I don't see a reason why we would want to restrict the hash function used in the signing capability of the node.  By having it work with any hash function, we leave the choice to the user (or more likely, DEX platform).  For instance, I believe Lightning-type HTLCs (which could be used for atomic swapping with Lightning BTC) use RIPEMD-160, whereas SHA-256 seems like a more generally useful and available hash function.

`OP_HASH160` and `OP_HASH256` are very Bitcoin-specific (since they hash twice), and thus not really something we should use here where we interface to another blockchain / network entirely.

Reply:
Alright. Sounds reasonable 👍